scripts/release.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+set -u
+
+if [ $# -lt 2 ]; then
+  echo "usage: $0 <previous-tag> <current-tag>"
+  exit 1
+fi
+
+github_url="https://github.com/areaDetector"
+
+
+# Find all tags in a given commit range (start_commit, end_commit] for a module
+get_module_releases() {
+  module=$1
+  start_commit=$2
+  end_commit=$3
+
+  # Take commit right after the first one
+  first_commit=$(git -C $module log --format=%h --reverse $start_commit..$end_commit | head -n 1)
+
+  tags="$(git -C $module tag --contains $first_commit --no-contains $end_commit)"
+
+  # Include `end_commit` to the range
+  tags+=" $(git -C $module tag --points-at $end_commit)"
+
+  echo $tags
+}
+
+# Print release information of each module based on the provided commit range
+# and releases
+print_module_releases() {
+  module=$1
+  start_commit=$2
+  end_commit=$3
+  releases="${@:4}"
+
+  read -ra tags <<< "$releases"
+
+  release_notes_url=$github_url/$module/blob/master/RELEASE.md
+  compare_url=$github_url/$module/compare/$start_commit..$end_commit
+
+  if [ ${#tags[@]} -gt 0 ]; then
+    echo "* [$module]($compare_url) [${tags[@]}]($release_notes_url#${tags[-1],,})"
```

Comment:
Note that here I'm using a pattern for the anchor that is not compatible with existing release notes from submodules. I'm assuming we could change to a simpler header which is only the tag.

It seems feasible to download the corresponding release file and get the header from that. But the same algorithm for reducing that into an anchor would need to be implemented here. I'm not sure if we'd like to do that.